src/toil/jobStores/fileJobStore.py

```diff
@@ -121,8 +122,18 @@ def batch(self):
             self.update(jobGraph)
         self._batchedJobGraphs = None
 
-    def exists(self, jobStoreID):
-        return os.path.exists(self._getJobFileName(jobStoreID))
+    def exists(self, jobStoreID,maxTries=1):
```

Comment:
I think it would be clearer to provide a separate function whose job it is to check for it a file that might appear vs just exists when called.  This caused some confusion with people looking at the code when this would happen.

Maybe waitForExists ?